weareinreach/InReach
# feat(IN-839) Hours display and Hours drawer

This PR covers refactoring the Hours display component as well as creating the Hours Drawer edit component.

Hours display component:
- Refactored to use 'interval' json string instead of start and end date time objects
- This component will display multiple time segments per day
- This component will display 'Open 24 Hours' if start and end time are the same and the 'Closed' property is set to false
- This component will display 'Closed' if the closed property is true for a specified dayIndex (ignores the interval value)

<img width="656" alt="Screen Shot 2023-08-03 at 5 33 46 PM" src="https://github.com/weareinreach/InReach/assets/10123216/0a3b77e9-23c0-4de6-bb79-f839d9849869">

Hours Drawer component:
- if the organization has hours data, that data will be displayed, which includes timezone
- The user can edit the timezone, that timezone will be applied to all day indexes and segments
- The user can edit existing start and end times (Open time, Close time)
- The user can add a new time segment for a day
- The user can set the day to Open 24hours
- The user can delete a time segment
- If a day has no time segment, the Closed property will be set to true

Hours Drawer Logic:
- if a user converts a day which has multi segments to 24 hours, the logic should set the first segment to 00:00, 00:00 and set the 'delete' property to true for all other segments

Validation:
- a segment must have a start and an end time
- if a day has multiple segments, those segments can't overlap
- sum of segment durations should not be equal to or over 24hours, if equal to 24 hours, suggest to use the Open 24hours checkbox instead

ToDo:
- if open 24 hours, set start and end time to be the same (hours display expects start and end to match, 00:00, 00:00)
- find a better library for the time input fields
- add ability to remove a time segment
- order segments based on earliest time first when listing
- add validation per segment
- add validation per hours object

# Pull Request type



Please check the type of change your PR introduces:


## What is the current behavior?



Issue Number: N/A

## What is the new behavior?



-
-
-

## Does this introduce a breaking change?




## Other information
